shapely/tests/test_io.py

```diff
@@ -190,14 +190,28 @@ def test_from_wkt_on_invalid(wkt, on_invalid, error, message):
         ("LINESTRING (0 0)", None),
         ("NOT A WKT STRING", None),
         ("POLYGON ((0 0, 0 0))", None),
+        ("POLYGON ((0 0, 1 1, 0 1))", "POLYGON ((0 0, 1 1, 0 1, 0 0))"),
         ("POLYGON ((0 0, 1 1))", "POLYGON ((0 0, 1 1, 0 0))"),
+        ("MULTIPOLYGON (((5 5, 6 6, 6 5, 5 5)), ((0 0, 0 0)))", None),
+        (
+            "MULTIPOLYGON (((5 5, 6 6, 6 5, 5 5)), ((0 0, 1 1)))",
+            "MULTIPOLYGON (((5 5, 6 6, 6 5, 5 5)), ((0 0, 1 1, 0 0)))",
+        ),
+        (
+            "GEOMETRYCOLLECTION (POLYGON ((5 5, 6 6, 6 5, 5 5)), POLYGON ((0 0, 0 0)))",
+            None,
+        ),
     ],
 )
 def test_from_wkt_on_invalid_fix(wkt, expected_wkt):
     """Tests for on_invalid="fix".
 
     Geometries that cannot be fixed are returned as None.
     """
+    # Make sure the wkt needs fixing
+    assert shapely.from_wkt(wkt, on_invalid="ignore") is None
```

Comment:
@theroggy Let's not use `from_wkt()` like this to pre-assert. If there were a bug involving `on_invalid="ignore"` we could be misled.

Reply:
I was actually trying to avoid being misled if the "fix" functionality of GEOS would be expanded to support more cases. 

But just a classic potato-patato case... so no strong opinion, so I'll remove it.